### DISCUSSION TITLE
What is the (qualitative) performance of different models?

### DISCUSSION THREAD

**Discussion by dfurrer** [blakeblackshear/frigate#6204] [General]
I'm trying to understand the quality of detections. What is the precision and recall e.g. for person detection? What model is used? 

I'm currently running on a Raspberry Pi. The docs say "It is recommended to use a hardware accelerated detector type instead for better performance." but based on context I assume that just means how many fps can be processed so if the Raspi can keep up there is no point in getting an EdgeTPU. Or are better/bigger models used if I connect an Edge TPU?

**Comment by NickM-27** [comment]
The concern with an AI accelerator is not only how many fps can be processed. There's a large reduction in CPU usage (which is a consideration especially on the pi), there's also a large latency advantage, and there's a power usage advantage as well. 

Of course it could mean a bigger model is run similarly, but by default frigate runs the same model for cpu and coral. 

Different models can have different training methods, different precision, model input size, etc which all affect its results.

**Reply by dfurrer** [reply]
> by default frigate runs the same model for cpu and coral.

Ok, I see. Investigating some more by reading the code here I see it's the "SSDLite MobileDet" model which comes as demonstration from Coral (mentioned here: https://coral.ai/models/object-detection/). So it's a MobileNetV2 architecture (https://arxiv.org/pdf/1801.04381.pdf, Mar 2019) which can do 32.9% mAP (320x320, 9.1ms on the coral)

Reading up some more, state of the art realtime models seem to be YOLO (e.g. v7):
https://paperswithcode.com/sota/real-time-object-detection-on-coco

The latest I can find on Edge TPU is Yolo v5 here:
https://github.com/guichristmann/edge-tpu-tiny-yolo (from these numbers it doesn't actually seem better than the current model though)

**Reply by NickM-27** [reply]
many models trained on the COCO dataset just won't be as good purely because they are trained on data that is not like the perspective of a security camera. That is the point of Frigate+ https://docs.frigate.video/integrations/plus which aims to provide better models
